File: languages/php/Bloock/Entity/Availability/IpfsPublisher.php
<?php

namespace Bloock\Entity\Availability;

use Bloock\DataAvailabilityType;

/**
 * Represents a publisher for IPFS data availability.
 */
class IpfsPublisher implements Publisher
{
    private int $type;
    private PublisherArgs $args;

    /**
     * Constructs a IpfsPublisher object with the specified parameters.
     */
    public function __construct()
    {
        $this->type = DataAvailabilityType::IPFS;
        $this->args = new PublisherArgs();
    }

    public function toProto(): \Bloock\Publisher
    {
        $p = new \Bloock\Publisher();
        $p->setType($this->type);
        $p->setArgs($this->args->toProto());
        return $p;
    }
}

File: languages/python/bloock/entity/availability/ipfs_publisher.py
import bloock._bridge.proto.bloock_availability_entities_pb2 as proto
from bloock.entity.availability.publisher import Publisher
from bloock.entity.availability.publisher_args import PublisherArgs


class IpfsPublisher(Publisher):
    """
    Represents a publisher for IPFS data availability.
    """
    def __init__(self) -> None:
        """
        Constructs a IpfsPublisher object with the specified parameters.
        :rtype: object
        """
        super().__init__(type=proto.IPFS, args=PublisherArgs())

    def to_proto(self) -> proto.Publisher:
        return proto.Publisher(type=self.type, args=self.args.to_proto())
